controllers/external-accounts.js

```diff
@@ -119,4 +120,95 @@ const syncExternalAccountData = async (req, res) => {
   }
 };
 
-module.exports = { addExternalAccountData, getExternalAccountData, syncExternalAccountData };
+/**
+ * Gets all group-roles
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+const newSyncExternalAccountData = async (req, res) => {
+  try {
+    const [discordUserData, unArchivedRdsUsersData] = await Promise.all([
+      getDiscordMembers(),
+      fetchUsersForKeyValues("roles.archived", false),
+    ]);
+    let usersArchivedCount = 0;
+    let usersUnArchivedCount = 0;
+    let totalUsersProcessed = unArchivedRdsUsersData.length;
+
+    const discordUserIdSet = new Set();
+
+    discordUserData.forEach((discordUser) => discordUserIdSet.add(discordUser.user.id));
+
+    let updateUserList = [];
+
+    for (const rdsUser of unArchivedRdsUsersData) {
+      let userData = {};
+
+      // This if-block will be removed if the IN_DISCORD ROLE is deprecated.
```

Comment:
Future-dated ticket please with a TODO comment

Reply:
Made the necessary changes.